compiler/rustc_lint/src/internal.rs

```diff
@@ -344,6 +354,13 @@ impl<'tcx> LateLintPass<'tcx> for TypeIr {
                 TypeIrInherentUsage,
             );
         }
+        if let Some(seg) = path.segments.iter().find(|seg| {
```

Comment:
So this only lints against direct *imports* of `rustc_type_ir`. It would be beneficial to also look for `TyKind::Path` and `ExprKind::Path` with path segments that resolve to `rustc_type_ir`.